homeassistant/helpers/llm.py

```diff
@@ -163,6 +171,22 @@ async def async_call(
         )
         return intent_response.as_dict()
 
+    @callback
+    def async_is_applicable(self, hass: HomeAssistant, tool_input: ToolInput) -> bool:
```

Comment:
When can this be used, we only have the tool_input after we fetch and pass the tools to the LLM?

Reply:
Actually everything except `tool_name` and `tool_args` is available at the time of getting the list of the tools. The idea is optionally pass the tool_input with available information to `async_get_tools`.